src/paint.rs

```diff
@@ -227,6 +235,18 @@ impl<'a> Painter<'a> {
                 output_buffer.push_str(&line);
                 output_buffer.push_str(ANSI_CSI_CLEAR_TO_EOL);
                 output_buffer.push_str(ANSI_SGR_RESET);
+            } else if line.is_empty() {
+                if let Some(empty_line_style) = empty_line_style {
+                    output_buffer.push_str(
+                        &empty_line_style
+                            .ansi_term_style
+                            .paint(format!(
+                                "{}{}",
+                                ANSI_CSI_CLEAR_TO_BOL, ANSI_CSI_CURSOR_BACK_1
+                            ))
+                            .to_string(),
```

Comment:
@phillipwood I've rebased over the commit on which you commented earlier but the call to `ansi_term_style.paint()` here wraps `\033[1K\033[1D` with a prefix which sets the background color and a `SGR RESET` suffix. Therefore I believe the end result is to emit the pattern of ANSI sequences that you suggested (modulo the details of the background color):

```
'\033[41m\033[1K\033[1D\033[m\n'
```

But do let me know if something doesn't look right!

Reply:
Ah that explains the reset. However I don't think you need `ANSI_CSI_CURSOR_BACK_1` though. Looking back at my comments on the issue I must have copied and pasted the wrong line from where I had been playing around in the terminal. It should have been `<set background color><ANSI_CLEAR_TO_BOL><ANSI_SGR_RESET><\n>` The `ANSI_CSI_CUROR_BACK_1` was left over from something else.  Sorry for the confusion

Reply:
Right, thanks for that. I have removed the `ANSI_CSI_CURSOR_BACK_1`.